app/controllers/test/CustomsDeclarationsStubController.scala

```diff
@@ -82,7 +83,15 @@ class CustomsDeclarationsStubController @Inject()(notificationService: Notificat
         <Details>[detail block]</Details>
       </Root>
 
-    notificationService.save(notification)
+    val cdsFileUploadConfig = appConfig.microservice.services.cdsFileUpload
+
+    val cdsFileUploadBaseUrl = cdsFileUploadConfig.protocol.get + "://" + cdsFileUploadConfig.host + ":" + cdsFileUploadConfig.port.get
```

Comment:
Is this string concatenation Java?  Why are protocol and port Options when they seem to be requred?

Reply:
This is Stub part. This service contains stub inside, I didn't spend a lot of time on this part.

I don't want to extract this to the Config file, because it's not something that should be used.

I don't like the idea to have stub inside the application but for now it's the easiest way to reuse implementation which was previously in the project.